### PR TITLE
Multi-repo: ensure external source gets updated if repo url changes

### PR DESCRIPTION
**Pull request by eysi09**

### CODE REVIEW

File: garden-cli/src/vcs/git.ts [hunk 1]
```diff
@@ -6,28 +6,26 @@
...
+  gitCli: (cwd: string): (...args: string[]) => Promise<string> => {
+    return async (...args) => {
+      return execa.stdout("git", args, { cwd })
     }
   },
 }
 
-function getUrlHash(url: string) {
-  return (parse(url).hash || "").split("#")[1]
+function getGitUrlParts(url: string): { repositoryUrl: string, hash: string } {
```
Comment (edvald) [inline garden-cli/src/vcs/git.ts]: Doesn't really matter here, but this type annotation can also just be inferred and not explicitly stated.

File: garden-cli/src/vcs/git.ts [hunk 2]
```diff
@@ -75,43 +74,43 @@ export class GitHandler extends VcsHandler {
...
-      const branch = hash ? `--branch=${hash}` : ""
-
-      await git(`clone --depth=1 ${branch} ${url} ${name}`)
+    const absPath = join(this.projectRoot, this.getRemoteSourcePath(name, url, sourceType))
+    const isCloned = await pathExists(absPath)
 
+    if (!isCloned) {
+      const entry = logEntry.info({ section: name, msg: `Fetching from ${url}`, entryStyle: EntryStyle.activity })
+      const { repositoryUrl, hash } = getGitUrlParts(url)
+      await git("clone", "--depth", "1", "--branch", hash, repositoryUrl, absPath)
```
Comment (edvald) [inline garden-cli/src/vcs/git.ts]: What if the hash is an empty string here?

File: garden-cli/src/vcs/git.ts [hunk 3]
```diff
@@ -75,43 +74,43 @@ export class GitHandler extends VcsHandler {
...
+    const { repositoryUrl, hash } = getGitUrlParts(url)
+
     await this.ensureRemoteSource({ url, name, sourceType, logEntry })
 
     const entry = logEntry.info({ section: name, msg: "Getting remote state", entryStyle: EntryStyle.activity })
-    await git("remote update")
+    await git("remote", "update")
+
+    const remoteHash = (await git("ls-remote", repositoryUrl, hash)).split("\t")[0]
+    const localHash = (await git("show-ref", "--hash", hash)).split("\n")[0]
```
Comment (edvald) [inline garden-cli/src/vcs/git.ts]: Same question here. Do we need to have a default hash/branch? Or would it make more sense to conditionally add these params?

**Comment by eysi09** [comment]
I've resolved the issues above.

**Pull request merged**